### ISSUE TITLE
oc new-app with multiple parameters regression

### ISSUE THREAD

**Issue by ewolinetz** [openshift/origin#8668]
I noticed this starting to happen today, if I have a template that i try to process and create using `oc new-app`

1) if there are no parameters to pass in, this works as expected
2) if there are multiple parameters, with different attempts of the same command (!!) I get different parameter names coming back as "unexpected"
I've experimented using multiple "-p" parameters and using a comma delimited list with single "-p"
I've also tried explicitly adding "--template=" before the template name

I can confirm the template is valid since `oc process {template} -v .... | oc create -f -` still works.

##### Version
openshift v1.3.0-alpha.0-169-g68d9241
kubernetes v1.3.0-alpha.1-331-g0522e63
etcd 2.3.0

##### Steps To Reproduce
1. Have a template file that has multiple parameters to set:
```
https://github.com/openshift/origin-aggregated-logging/blob/master/deployment/templates/support.yaml
```
2. Run oc new-app with providing multiple parameters: 
```
oc new-app -f templates/support.yaml --param OAUTH_SECRET=HKS3jTwLQHdEOj0WrOlx1Z71xTYHQAlktcalH3m0tYwo4LOEsKoDcVs703Vr3W9S --param KIBANA_HOSTNAME=kibana.example.com --param KIBANA_OPS_HOSTNAME=kibana-ops.example.com --param IMAGE_PREFIX_DEFAULT=172.30.192.225:5000/logging/
```
##### Current Result
Different parameters are coming back as "unexpected parameter name"
`error: unexpected parameter name "OAUTH_SECRET"`

##### Expected Result
The template should be processed and the objects should be created

**Comment by sosiouxme** [comment]
Might be related to b591700ee0492328d281a201974557985ee32621 ?

**Comment by aweiteka** [comment]
I was able to repro this.

FWIW, my workaround is to process the template and pass to create.

```
oc process <template_name> -v FOO=BAR,BAR=FOO | oc create -f -
```

**Comment by smarterclayton** [comment]
I'm getting: error: unable to load template file "https://github.com/openshift/origin-aggregated-logging/blob/master/deployment/templates/support.yaml": bufio.Scanner: token too long

on that template.

**Comment by sosiouxme** [comment]
try the raw version of that page

**Comment by smarterclayton** [comment]
It's not finding the parameter in the template - can you think of why that is?

**Comment by aweiteka** [comment]
I'm not at all familiar with the code path but does it help that oc process is working?

**Comment by sosiouxme** [comment]
The parameter is in the template. It's just not finding it.

**Comment by smarterclayton** [comment]
It's fixed in #8674.  Added a test case, which we did not have.
